src/services/ieq/ieq.js

```diff
@@ -387,25 +387,29 @@ IEQ.prototype.closeSubscriptions = function(){
 
 /**
 * Request Data to make CSV file
-	* @param {list} sensorNames : list of sensor and index names
-	* @param {number} _firstDay: timestamp of beginning time
-  	* @param {string} timeSample: timeinterval for data. Parameters: "second", "minute", "hour", "day", "week", "month"
-	* @param {number} _nlines: maximum number of lines requested
-	* @param {callback} callback: called after update
+	 @param {object} csvConfig params:
+		* @param {list} sensorNames : list of sensor and index names
+		* @param {number} _startTime: timestamp of beginning time
+		* @param {number} _endTime: timestamp of end time
+	  	* @param {string} timeSample: timeinterval for data. Parameters: "second", "minute", "hour", "day", "week", "month"
+		* @param {number} _nlines: maximum number of lines requested
+		* @param {callback} callback: called after update
 */
 
 
-IEQ.prototype.getCSVData = function(sensorNames,_firstDay, timeSample ,_nlines, callback){
+IEQ.prototype.getCSVData = function(csvConfig, callback){
 	var that = this;
-	var firstDay = new Date(_firstDay);
+
+	if (csvConfig && typeof csvConfig.nlines !="number" ) csvConfig.nlines = undefined;
+
 	var dataConfig = {
 		criteria: {
-			time: { start: firstDay.getTime(), rangeUnit: 'hour', range: 180, sampling: timeSample}, // 360h -> 15d // 180h -> 7j
+			time: { start: csvConfig.startTime, end: csvConfig.endTime , sampling:csvConfig.timeSample},
```

Comment:
I really would rather parse startTime and endTime in a new Date().getTime() instead of assuming that SDK user will comply to sending a timestamp in ms.
Note that new Date(timestamp) would still work.